libkbfs/block_types.go

```diff
@@ -105,17 +106,17 @@ func NewDirBlock() Block {
 
 // NewEmpty implements the Block interface for DirBlock
 func (db *DirBlock) NewEmpty() Block {
-	return NewDirBlock()
+	return &DirBlock{}
 }
 
 // Set implements the Block interface for DirBlock
 func (db *DirBlock) Set(other Block, codec kbfscodec.Codec) {
 	otherDb := other.(*DirBlock)
-	copy, err := otherDb.DeepCopy(codec)
+	err := kbfscodec.Update(codec, db, otherDb)
```

Comment:
Hmm, ok.  Seems like we might be able to fix that some other way, by initializing/not initializing those slices in other places, but if that's too hard for this PR, can you please explain why DeepCopy isn't used in a comment here?  Thanks!